### DISCUSSION TITLE
Signup/Signin using unique link (not magic link)

### DISCUSSION THREAD

**Discussion by viraj-lead** [supabase/supabase#30987] [Questions]
I currently have only magic links enabled in my app. For some of my enterprise customers, the magic link never delivers or takes a long time to reach their inbox. I need to get away from email deliverability issues and was wondering if there was a way to generate a magic link of some sort locally for a given user email that I can send to a customer directly(with a link validity ofcourse). As far as security is concerned, I know magic link delivered to an inbox will be safer but I am okay to take a hit on this.

P.S: I cannot enable social or any other Oauth login and am looking for a login via link option which is not delivered to email.

**Comment by GaryAustin1** [comment]
You can use the auth admin https://supabase.com/docs/reference/javascript/auth-admin-generatelink (generateLink) to get the OTP code and give that to them directly to enter into a page that does verifyOtp.   Also the Otp in an email might get thru firewalls better than ones with a link in them.
